src/gpu/intel/sycl/l0/utils.cpp

```diff
@@ -272,6 +287,114 @@ bool compare_ze_devices(const ::sycl::device &lhs, const ::sycl::device &rhs) {
+            = ZE_DEVICE_FP_ATOMIC_EXT_FLAG_GLOBAL_LOAD_STORE
+            | ZE_DEVICE_FP_ATOMIC_EXT_FLAG_LOCAL_LOAD_STORE;
+    ze_device_fp_atomic_ext_flags_t atomic_add
+            = ZE_DEVICE_FP_ATOMIC_EXT_FLAG_GLOBAL_ADD
+            | ZE_DEVICE_FP_ATOMIC_EXT_FLAG_LOCAL_ADD;
+    ze_device_fp_atomic_ext_flags_t atomic_min_max
+            = ZE_DEVICE_FP_ATOMIC_EXT_FLAG_GLOBAL_MIN_MAX
+            | ZE_DEVICE_FP_ATOMIC_EXT_FLAG_LOCAL_MIN_MAX;
+
+    if ((fltAtom.fp16Flags & atomic_load_store) == atomic_load_store)
+        native_extensions |= (uint64_t)native_ext_t::fp16_atomic_load_store;
+    if ((fltAtom.fp16Flags & atomic_add) == atomic_add)
+        native_extensions |= (uint64_t)native_ext_t::fp16_atomic_add;
+    if ((fltAtom.fp16Flags & atomic_add) == atomic_min_max)
+        native_extensions |= (uint64_t)native_ext_t::fp16_atomic_min_max;
+
+    if ((fltAtom.fp32Flags & atomic_load_store) == atomic_load_store)
+        native_extensions |= (uint64_t)native_ext_t::fp32_atomic_load_store;
+    if ((fltAtom.fp32Flags & atomic_add) == atomic_add)
+        native_extensions |= (uint64_t)native_ext_t::fp32_atomic_add;
+    if ((fltAtom.fp32Flags & atomic_add) == atomic_min_max)
+        native_extensions |= (uint64_t)native_ext_t::fp32_atomic_min_max;
+
+    if ((fltAtom.fp64Flags & atomic_load_store) == atomic_load_store)
+        native_extensions |= (uint64_t)native_ext_t::fp64_atomic_load_store;
+    if ((fltAtom.fp64Flags & atomic_add) == atomic_add)
+        native_extensions |= (uint64_t)native_ext_t::fp64_atomic_add;
+    if ((fltAtom.fp64Flags & atomic_add) == atomic_min_max)
+        native_extensions |= (uint64_t)native_ext_t::fp64_atomic_min_max;
+
+    return status::success;
+}
+
+status_t get_l0_device_eu_count(ze_device_handle_t device, int &eu_count) {
+    ze_device_properties_t deviceProps = {ZE_STRUCTURE_TYPE_DEVICE_PROPERTIES};
+    ze_eu_count_ext_t eucnt = ze_eu_count_ext_t();
+    deviceProps.pNext = &eucnt;
+
+    CHECK(func_zeDeviceGetProperties(device, &deviceProps));
+    eu_count = eucnt.numTotalEUs;
+    return status::success;
+}
+
+status_t init_gpu_hw_info(impl::engine_t *engine, ze_device_handle_t device,
+        ze_context_handle_t context, uint32_t &ip_version,
+        compute::gpu_arch_t &gpu_arch, int &gpu_product_family,
+        int &stepping_id, uint64_t &native_extensions, bool &mayiuse_systolic,
+        bool &mayiuse_ngen_kernels) {
+    using namespace ngen;
+    HW hw = HW::Unknown;
+    Product product = {ProductFamily::Unknown, 0};
+    LevelZeroCodeGenerator<HW::Unknown>::detectHWInfo(
+            context, device, hw, product);
+
+    gpu_arch = jit::convert_ngen_arch_to_dnnl(hw);
+    gpu_product_family = static_cast<int>(product.family);
+    stepping_id = product.stepping;
+
+    mayiuse_systolic = false;
+    CHECK(get_l0_device_enabled_systolic_intel(device, mayiuse_systolic));
+    // TODO: xelpg has no f64 support. check that the query properly handle that
```

Comment:
Did you check this? :)

Reply:
Yes it was checked, I will drop the comment. :)